### PR TITLE
Replace MatLegacyDialog with MatDialog

### PR DESCRIPTION
**Pull request by MoritzWeber0**

### CODE REVIEW

File: frontend/src/app/sessions/active-sessions/file-browser/file-browser.component.html [hunk 1]
```diff
@@ -111,7 +111,7 @@ <h1>File browser (experimental)</h1>
   </mat-nested-tree-node>
 </mat-tree>
 
-<mat-dialog-actions>
+<div>
```
Comment (dominik003) [inline frontend/src/app/sessions/active-sessions/file-browser/file-browser.component.html:114]: The file browser component definitely misses some padding right now. So I think wrapping everything in a `div` with the `dialog` class would be sufficient here.

File: frontend/src/app/sessions/new-readonly-session-dialog/new-readonly-session-dialog.component.html [hunk 2]
```diff
@@ -4,17 +4,17 @@
  -->
 
 <form (ngSubmit)="requestSession()">
```
Comment (dominik003) [inline frontend/src/app/sessions/new-readonly-session-dialog/new-readonly-session-dialog.component.html:6]: In the `new-readonly-session-dialog.component.ts` the old `legacy-dialog` is still used and therefore requesting a readonly session is not possible right now. In addition, you need to add the `dialog` class here to add some padding to the dialog.

File: frontend/src/app/projects/models/backup-settings/view-logs-dialog/view-logs-dialog.component.html [hunk 3]
```diff
@@ -3,14 +3,14 @@
  ~ SPDX-License-Identifier: Apache-2.0
  -->
 
-<div class="wrapper">
+<div class="wrapper dialog">
```
Comment (dominik003) [inline frontend/src/app/projects/models/backup-settings/view-logs-dialog/view-logs-dialog.component.html:6]: In the `view-logs-dialog.component.ts` the legacy dialog is still used.

File: frontend/src/app/settings/core/tools-settings/tool-details/tool-deletion-dialog/tool-deletion-dialog.component.html [hunk 4]
```diff
@@ -3,8 +3,8 @@
  ~ SPDX-License-Identifier: Apache-2.0
  -->
 
-<h1 mat-dialog-title>Remove Tool</h1>
-<div mat-dialog-content>
+<h1>Remove Tool</h1>
+<div>
```
Comment (dominik003) [inline frontend/src/app/settings/core/tools-settings/tool-details/tool-deletion-dialog/tool-deletion-dialog.component.html:7]:
```suggestion
<div class="dialog">
```
Similar to other dialogs, this dialogs misses some padding which should be solved by adding the `dialog` class to the div here.

File: frontend/src/app/settings/modelsources/git-settings/delete-git-settings-dialog/delete-git-settings-dialog.component.html [hunk 5]
```diff
@@ -3,15 +3,15 @@
  ~ SPDX-License-Identifier: Apache-2.0
  -->
 
-<h1 mat-dialog-title>Delete Instance {{ instance.name }}</h1>
-<div mat-dialog-content>
+<h1>Delete Instance {{ instance.name }}</h1>
```
Comment (dominik003) [inline frontend/src/app/settings/modelsources/git-settings/delete-git-settings-dialog/delete-git-settings-dialog.component.html:6]: This dialog misses some padding, so please put a div with the `dialog` class around it.

File: frontend/src/app/settings/modelsources/t4c-settings/t4c-instance-settings/t4c-repo-deletion-dialog/t4c-repo-deletion-dialog.component.html [hunk 6]
```diff
@@ -3,8 +3,8 @@
  ~ SPDX-License-Identifier: Apache-2.0
  -->
 
-<h1 mat-dialog-title>Remove T4C repository</h1>
```
Comment (dominik003) [inline frontend/src/app/settings/modelsources/t4c-settings/t4c-instance-settings/t4c-repo-deletion-dialog/t4c-repo-deletion-dialog.component.html:6]: Please add a `div` with the `dialog` class here.

**Review by dominik003 — CHANGES_REQUESTED** [review]
I think I have tried all the dialogs except the View Log dialog and added a comment to the dialogs where a padding/the dialog class is missing. Hopefully, I didn't miss one but in the worst case we have a dialog without a padding.

File: frontend/src/app/projects/models/backup-settings/create-backup/create-backup.component.html [hunk 7]
```diff
@@ -3,71 +3,72 @@
...
-  </button>
+  <div class="flex-space-between">
+    <button mat-button [mat-dialog-close]="false">Cancel</button>
+    <button
+      mat-button
+      *ngIf="t4cAndGitModelExists"
+      (click)="createGitBackup()"
+      cdkFocusInitial
+      [disabled]="!createBackupForm.valid"
+    >
```
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply frontend/src/app/projects/models/backup-settings/create-backup/create-backup.component.html:70]: Done.

**Review by dominik003 — APPROVED** [review]
It only needs a rebase and can then be merged

**Pull request merged**